cloudinit/config/schemas/schema-network-config-v1.json

```diff
@@ -0,0 +1,562 @@
+                }
+              }
+            },
+            "subnets": {
+              "type": "array",
+              "items": {
+                  "$ref": "#/$defs/config_type_subnet"
+              }
+            }
+          }
+        },
+        "anyOf_type_vlan": {
+          "type": "object",
+          "additionalProperties": false,
+          "required": ["type", "name", "vlan_link", "vlan_id"],
+          "properties": {
+            "type": {
+              "type": "string",
+              "enum": ["vlan"]
+            },
+            "name": {
+              "type": "string",
+              "description": "Name of the VLAN."
+            },
+            "vlan_link": {
+              "type": "string",
+              "description": "Specify the underlying link via its ``name``."
+            },
+            "vlan_id": {
+              "type": "integer",
+              "description": "Specify VLAN numeric id."
+            },
+            "mtu": {
+              "type": "integer",
+              "description": "The MTU size in bytes. The ``mtu`` key represents a device's Maximum Transmission Unit, which is the largest size packet or frame, specified in octets (eight-bit bytes), that can be sent in a packet- or frame-based network. Specifying ``mtu`` is optional. Values too small or too large for a device may be ignored by that device."
+            },
+            "subnets": {
+              "type": "array",
+              "items": {
+                  "$ref": "#/$defs/config_type_subnet"
+              }
+            }
+          }
+        },
+        "anyOf_type_nameserver": {
+          "type": "object",
+          "additionalProperties": false,
+          "required": ["type", "address", "search"],
+          "properties": {
+            "type": {
+              "type": "string",
+              "enum": ["nameserver"]
+            },
+            "address": {
+              "description": "List of IPv4 or IPv6 address of nameservers.",
+              "type": "array",
+              "items": {
+                "type": "string"
+              }
+            },
+            "search": {
+              "description": "List of hostnames to include in the `resolv.conf` search path.",
+              "type": "array",
+              "items": {
+                "type": "string"
+              }
+            },
+            "interface": {
+              "type": "string",
+              "description": "Optional. Ties the nameserver definition to the specified interface. The value specified here must match the ``name`` of an interface defined in this config. If unspecified, this nameserver will be considered a global nameserver."
+            }
+          }
+        },
+        "anyOf_type_route": {
+          "type": "object",
+          "additionalProperties": false,
+	  "anyOf": [
+	      {"required": ["network", "gateway"]},
```

Comment:
Network or destination keys comes from https://github.com/canonical/cloud-init/blob/main/cloudinit/net/network_state.py#L1051-L1070 which will [raise ValueError if neither destination nor network is present](https://github.com/canonical/cloud-init/blob/main/cloudinit/net/network_state.py#L992-#L998)

Reply:
Wihle it would be nice to make it required the python logic itself allows for no `type: route` definition to cut down on repetition when as the the parent key scope is typically `routes:`.
- [code comments suggest the optional nature of `type` key](url)
- OLD SRU unittests/integration/verification tests[[1](https://github.com/canonical/cloud-init/blob/main/tests/unittests/cmd/devel/test_net_convert.py#L204-L208)], [[2](https://github.com/canonical/cloud-init/blob/main/tests/unittests/test_net.py#L2253-L2255)], [[3](https://github.com/cloud-init/ubuntu-sru/blob/main/bugs/lp-openstack-ipv6-slaac-dhcpv6.txt#L798-L800)] also assert success without the `type: route` dict item.